### PR TITLE
FINERACT-1815: Fix datatable put call with resource id

### PR DESCRIPTION
**Pull request by josehernandezfintecheandomx**
## Description

Update Datatable One to Many Entry `PUT` for:

`resource_id` field in response is populated with `datatableEntryId`

[FINERACT-1815](https://issues.apache.org/jira/browse/FINERACT-1815)

## Checklist

Please make sure these boxes are checked before submitting your pull request - thanks!

- [ ] Write the commit message as per https://github.com/apache/fineract/#pull-requests

- [ ] Acknowledge that we will not review PRs that are not passing the build _("green")_ - it is your responsibility to get a proposed PR to pass the build, not primarily the project's maintainers.

- [ ] Create/update unit or integration tests for verifying the changes made.

- [ ] Follow coding conventions at https://cwiki.apache.org/confluence/display/FINERACT/Coding+Conventions.

- [ ] Add required Swagger annotation and update API documentation at fineract-provider/src/main/resources/static/legacy-docs/apiLive.htm with details of any API changes

- [ ] Submission is not a "code dump".  (Large changes can be made "in repository" via a branch.  Ask on the developer mailing list for guidance, if required.)

FYI our guidelines for code reviews are at https://cwiki.apache.org/confluence/display/FINERACT/Code+Review+Guide.

### CODE REVIEW

File: fineract-provider/src/main/java/org/apache/fineract/infrastructure/dataqueries/service/ReadWriteNonCoreDataServiceImpl.java [hunk 1]
```diff
@@ -1415,14 +1415,22 @@ private CommandProcessingResult updateDatatableEntry(final String dataTableName,
...
                 .withGroupId(commandProcessingResult.getGroupId()) //
                 .withClientId(commandProcessingResult.getClientId()) //
                 .withSavingsId(commandProcessingResult.getSavingsId()) //
                 .withLoanId(commandProcessingResult.getLoanId()) //
-                .with(changes) //
-                .build();
+                .with(changes);
+
+        if (command.subentityId() != null) {
+            result = result.withEntityId(command.subentityId());
```
Comment (galovics) [inline fineract-provider/src/main/java/org/apache/fineract/infrastructure/dataqueries/service/ReadWriteNonCoreDataServiceImpl.java]: Why don't we simply set the subResourceId in this case?

Comment (josehernandezfintecheandomx) [reply fineract-provider/src/main/java/org/apache/fineract/infrastructure/dataqueries/service/ReadWriteNonCoreDataServiceImpl.java]: @galovics what you mean? Currently there is not any method in the `JsonCommand` class to return the `subresourceId` attribute, I mean the `subentityId()` method returns that attribute 

Screenshot 2022-11-21 at 13 25 02

Comment (adamsaghy) [reply fineract-provider/src/main/java/org/apache/fineract/infrastructure/dataqueries/service/ReadWriteNonCoreDataServiceImpl.java:1428]: I think we dont need these conditions.
`
CommandProcessingResultBuilder result = new CommandProcessingResultBuilder()
.withEntityId(command.entityId()
.withSubEntityId(command.subentityId()
...
`

Comment (adamsaghy) [reply fineract-provider/src/main/java/org/apache/fineract/infrastructure/dataqueries/service/ReadWriteNonCoreDataServiceImpl.java]: subEntity == subResourceId

Comment (adamsaghy) [reply fineract-provider/src/main/java/org/apache/fineract/infrastructure/dataqueries/service/ReadWriteNonCoreDataServiceImpl.java]: Personally i would do this:
```
new CommandProcessingResultBuilder().withEntityId(datatableId != null? datatableId : appTableId);
```

Comment (josehernandezfintecheandomx) [reply fineract-provider/src/main/java/org/apache/fineract/infrastructure/dataqueries/service/ReadWriteNonCoreDataServiceImpl.java]: Done

**Comment by galovics** [comment]
@josehernandezfintecheandomx can you pls change the commit message & PR title to have a Fineract ticket in it? Thanks.

**Comment by josehernandezfintecheandomx** [comment]
> @josehernandezfintecheandomx can you pls change the commit message & PR title to have a Fineract ticket in it? Thanks.

@galovics Title updated

**Review by galovics — APPROVED** [review]

**Pull request merged**